Using Hwloc failed

I'm getting this warning in the dev branch of Soss.jl:

```julia
julia> using Soss
┌ Warning: AssertionError("obj.type_ >= 0 && obj.type_ < length(obj_types)")
└ @ VectorizationBase ~/.julia/packages/VectorizationBase/bYkVf/src/topology.jl:9
┌ Warning:     Using Hwloc failed. Please file an issue with the above warning at: https://github.com/JuliaParallel/Hwloc.jl
│     Proceeding with generic topology assumptions. This may result in reduced performance.
└ @ VectorizationBase ~/.julia/packages/VectorizationBase/bYkVf/src/topology.jl:10
```

version:
```julia
julia> versioninfo()
Julia Version 1.6.0-rc1
Commit a58bdd9010 (2021-02-06 15:49 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: AMD Ryzen Threadripper 2950X 16-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, znver1)
Environment:
  JULIA_NUM_THREADS = 16
  JULIA_EDITOR = "/opt/visual-studio-code/code"
```

Relevant Manifest.jl lines:
```
[[Hwloc]]
deps = ["Hwloc_jll"]
git-tree-sha1 = "2e3d1d4ab0e7296354539b2be081f71f4b694c0b"
uuid = "0e44f5e4-bd66-52a0-8798-143a42290a1d"
version = "1.2.0"

[[Hwloc_jll]]
deps = ["Artifacts", "JLLWrappers", "Libdl", "Pkg"]
git-tree-sha1 = "1179250d910c99810d8a7ff55c50c4ed68c77a58"
uuid = "e33a78d0-f292-5ffc-b300-72abe9b543c8"
version = "2.4.0+0"
```

And outside of Soss:
```julia
julia> import Hwloc

julia> topology = Hwloc.topology_load()
ERROR: AssertionError: obj.type_ >= 0 && obj.type_ < length(obj_types)
Stacktrace:
 [1] load(hobj::Ptr{Hwloc.hwloc_obj_v2})
   @ Hwloc ~/.julia/packages/Hwloc/0st5s/src/Hwloc.jl:356
 [2] load(hobj::Ptr{Hwloc.hwloc_obj_v2})
   @ Hwloc ~/.julia/packages/Hwloc/0st5s/src/Hwloc.jl:377
 [3] topology_load()
   @ Hwloc ~/.julia/packages/Hwloc/0st5s/src/Hwloc.jl:342
 [4] top-level scope
   @ REPL[7]:1
```

@vchuravy suggested adding a `@show`, which gives
```julia
julia> topology = Hwloc.topology_load()
obj.type_ = 0
obj_types = [:Machine, :Package, :Core, :PU, :L1Cache, :L2Cache, :L3Cache, :L4Cache, :L5Cache, :I1Cache, :I2Cache, :I3Cache, :Group, :NUMANode, :Bridge, :PCI_Device, :OS_Device, :Misc, :Error]
obj.type_ = 1
obj_types = [:Machine, :Package, :Core, :PU, :L1Cache, :L2Cache, :L3Cache, :L4Cache, :L5Cache, :I1Cache, :I2Cache, :I3Cache, :Group, :NUMANode, :Bridge, :PCI_Device, :OS_Device, :Misc, :Error]
obj.type_ = 19
obj_types = [:Machine, :Package, :Core, :PU, :L1Cache, :L2Cache, :L3Cache, :L4Cache, :L5Cache, :I1Cache, :I2Cache, :I3Cache, :Group, :NUMANode, :Bridge, :PCI_Device, :OS_Device, :Misc, :Error]
ERROR: AssertionError: obj.type_ >= 0 && obj.type_ < length(obj_types)
Stacktrace:
 [1] load(hobj::Ptr{Hwloc.hwloc_obj_v2})
   @ Hwloc ~/git/Hwloc.jl/src/Hwloc.jl:357
 [2] load(hobj::Ptr{Hwloc.hwloc_obj_v2})
   @ Hwloc ~/git/Hwloc.jl/src/Hwloc.jl:378
 [3] topology_load()
   @ Hwloc ~/git/Hwloc.jl/src/Hwloc.jl:342
 [4] top-level scope
   @ REPL[1]:1
```

--- a/Manifest.toml
+++ b/Manifest.toml
@@ -1,5 +1,11 @@
 # This file is machine-generated - editing it directly is not advised
 
+[[Artifacts]]
+deps = ["Pkg"]
+git-tree-sha1 = "c30985d8821e0cd73870b17b0ed0ce6dc44cb744"
+uuid = "56f22d72-fd6d-98f1-02f0-08ddc0907c33"
+version = "1.3.0"
+
 [[Base64]]
 uuid = "2a0f44e3-6c83-55bd-87e4-b1978d98bd5f"
 
@@ -7,21 +13,23 @@ uuid = "2a0f44e3-6c83-55bd-87e4-b1978d98bd5f"
 deps = ["Printf"]
 uuid = "ade2ca70-3891-5945-98fb-dc099432e06a"
 
-[[Distributed]]
-deps = ["Random", "Serialization", "Sockets"]
-uuid = "8ba89e20-285c-5b6f-9357-94700520ee1b"
-
 [[Hwloc_jll]]
-deps = ["Libdl", "Pkg"]
-git-tree-sha1 = "d9de29482e0a9efb0639328e208d02e01554fa9b"
+deps = ["Artifacts", "JLLWrappers", "Libdl", "Pkg"]
+git-tree-sha1 = "1179250d910c99810d8a7ff55c50c4ed68c77a58"
 uuid = "e33a78d0-f292-5ffc-b300-72abe9b543c8"
-version = "2.2.0+0"
+version = "2.4.0+0"
 
 [[InteractiveUtils]]
 deps = ["Markdown"]
 uuid = "b77e0a4c-d291-57a0-90e8-8db25a27a240"
 
+[[JLLWrappers]]
+git-tree-sha1 = "a431f5f2ca3f4feef3bd7a5e94b8b8d4f2f647a0"
+uuid = "692b3bcd-3c85-4b1f-b108-f13ce0eb3210"
+version = "1.2.0"
+
 [[LibGit2]]
+deps = ["Printf"]
 uuid = "76f85450-5226-5b5a-8eaa-529ad045b433"
 
 [[Libdl]]
@@ -35,7 +43,7 @@ deps = ["Base64"]
 uuid = "d6f4376e-aef5-505a-96c1-9c027394607a"
 
 [[Pkg]]
-deps = ["Dates", "LibGit2", "Libdl", "Logging", "Markdown", "Printf", "REPL", "Random", "SHA", "Test", "UUIDs"]
+deps = ["Dates", "LibGit2", "Libdl", "Logging", "Markdown", "Printf", "REPL", "Random", "SHA", "UUIDs"]
 uuid = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
 
 [[Printf]]
@@ -58,10 +66,6 @@ uuid = "9e88b42a-f829-5b0c-bbe9-9e923198166b"
 
 [[Sockets]]
 uuid = "6462fe0b-24de-5631-8697-dd941f90decc"
-
-[[Test]]
-deps = ["Distributed", "InteractiveUtils", "Logging", "Random"]
-uuid = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [[UUIDs]]
 deps = ["Random", "SHA"]

--- a/Project.toml
+++ b/Project.toml
@@ -1,7 +1,7 @@
 name = "Hwloc"
 uuid = "0e44f5e4-bd66-52a0-8798-143a42290a1d"
 authors = ["Erik Schnetter <schnetter@gmail.com>"]
-version = "1.2.0"
+version = "1.3.0"
 
 [deps]
 Hwloc_jll = "e33a78d0-f292-5ffc-b300-72abe9b543c8"

--- a/src/Hwloc.jl
+++ b/src/Hwloc.jl
@@ -40,7 +40,7 @@ const obj_types_v1 =
 const obj_types_v2 =
     Symbol[:Machine, :Package, :Core, :PU, :L1Cache, :L2Cache, :L3Cache,
            :L4Cache, :L5Cache, :I1Cache, :I2Cache, :I3Cache, :Group, :NUMANode,
-           :Bridge, :PCI_Device, :OS_Device, :Misc, :Error]
+           :Bridge, :PCI_Device, :OS_Device, :Misc, :MemCache, :Die, :Error]
 const obj_types = api_version < v"2" ? obj_types_v1 : obj_types_v2
 const cache_types =
     Symbol[:Unified, :Data, :Instruction]
@@ -278,6 +278,15 @@ function show(io::IO, a::OSDevAttr)
     print(io, "OSDev{type=$(string(a.type_))}")
 end
 
+mutable struct DieAttr <: Attribute
+    depth::Int
+end
+function show(io::IO, a::DieAttr)
+    print(io, "Die{depth=$(a.depth)}")
+end
+
+# type MemCacheAttr <: Attribute end
+
 
 
 mutable struct Object
@@ -410,6 +419,11 @@ function load_attr(hattr::Ptr{Cvoid}, type_::Symbol)
     elseif type_==:PCI_Device
         error("not implemented")
     elseif type_==:OS_Device
+        error("not implemented")
+    elseif type_==:Die
+        ha = unsafe_load(convert(Ptr{hwloc_cache_attr_s}, hattr))
+        return DieAttr(ha.depth)
+    elseif type_==:MemCache
         error("not implemented")
     else
         error("Unsupported object type $type_")

